packages/client-sdk-nodejs/src/internal/grpc/middlewares-interceptor.ts

```diff
@@ -75,6 +77,17 @@ export function middlewaresInterceptor(
             status: StatusObject,
             next: (status: StatusObject) => void
           ): void {
+            if (status.code === 4) {
+              const logger = loggerFactory.getLogger('grpc-interceptor');
+              logger.debug(
+                `Received status: ${status.code} ${
+                  status.details
+                } and grpc connection status: ${
+                  grpcClient?.getChannel()?.getConnectivityState(false) ??
```

Comment:
nice. can you maybe test this by setting a super low timeout in your local code? and maybe combine that with a `set` operation that sends a really large chunk of data?

Reply:
That worked! Got these logs in response:

```
[2024-02-03T00:55:41.292Z] DEBUG (Momento: FixedCountRetryStrategy): Determining whether request is eligible for retry; status code: 4, request type: /cache_client.Scs/Set, attemptNumber: 1, maxAttempts: 3
[2024-02-03T00:55:41.292Z] DEBUG (Momento: DefaultEligibilityStrategy): Response with status code 4 is not retryable.
[2024-02-03T00:55:41.292Z] DEBUG (Momento: RetryInterceptor): Request not eligible for retry: path: /cache_client.Scs/Set; response status code: 4.
[2024-02-03T00:55:41.292Z] DEBUG (Momento: grpc-interceptor): Received status: 4 Deadline exceeded and grpc connection status: 2
Error setting key: The client's configured timeout was exceeded; you may need to use a Configuration with more lenient timeouts: 4 DEADLINE_EXCEEDED: Deadline exceeded
[2024-02-03T00:55:41.303Z] DEBUG (Momento: FixedCountRetryStrategy): Determining whether request is eligible for retry; status code: 4, request type: /cache_client.Scs/Get, attemptNumber: 1, maxAttempts: 3
[2024-02-03T00:55:41.303Z] DEBUG (Momento: DefaultEligibilityStrategy): Response with status code 4 is not retryable.
[2024-02-03T00:55:41.303Z] DEBUG (Momento: RetryInterceptor): Request not eligible for retry: path: /cache_client.Scs/Get; response status code: 4.
[2024-02-03T00:55:41.303Z] DEBUG (Momento: grpc-interceptor): Received status: 4 Deadline exceeded and grpc connection status: 2
Error: The client's configured timeout was exceeded; you may need to use a Configuration with more lenient timeouts: 4 DEADLINE_EXCEEDED: Deadline exceeded
```